src/js/structs/CompositeState.js

```diff
@@ -100,8 +102,20 @@ class CompositeState {
     });
   }
 
+  set data(newData) {
+    this._data = newData;
+    this.updateNodesList();
+  }
+  get data() {
+    return this._data;
+  }
+
+  updateNodesList() {
+    this.nodesList = new NodesList({ items: this.data.slaves });
```

Comment:
Let's just reset the node list and the create the list in the respective getter to defer the creation as illustrated below:

```js

set data(newData) {
 this._data = newData;
 this._nodesList = null;
}

getNodesList() {
 if(this._nodesList === null){
     this._nodesList = new NodesList({ items: this.data.slaves });
  }

return this._nodesList;
}
```